### ISSUE TITLE
Scroll reveal not working correctly 

### ISSUE THREAD

**Issue by camsarmiento** [scrollreveal/scrollreveal#462]
Hey I'm using scroll reveal v4 at this website http://camilomancilla.com.co/AGR.html but with no reason stopped working, when scroll down shows the items for a while but then the items hide again as quick as I keep scrolling down, I've been searching about this bug but there isn't any case about this... Help me!!!

### Environment

* Operating System: Mac OS
* Browser Version: Chrome 69.0.3497.100 
* ScrollReveal Version: v4

**Comment by JohnPhilosopher** [comment]
I added `html, body {height:100%}` and it seems to work fine after that.

**Comment by jlmakes** [comment]
ScrollReveal 4 adds `body { height: 100%; }` automatically; historically applying this style to `html` has not been necessary, but I'm curious if it was required in this case.

Did you solve this @camsarmiento ?

**Comment by camsarmiento** [comment]
Not at all, I'be trying different stuff but it's been quite a mission, is there something else I can do ?

**Comment by JohnPhilosopher** [comment]
Looking further into it, weirdly if the width of the window is smaller than a certain amount it stops working. However, bigger screen and it starts working again. Below is a gif of what happened to me. From my experimentation, it might be a styling issue. However, I will look further.

https://www.dropbox.com/s/k44hb4dla2hd1a5/prototype_unfinished.gif

**Comment by twister65** [comment]
I had the same problem with my website, I called the sync function beforeReset to resolve it (with the reset option set to true). For example : 
sr.reveal('#search', { <?php echo $this->params->get('animation_search_text'); ?>, beforeReset:function(){sr.sync()} });
https://scrollrevealjs.org/api/sync.html
Note : this problem occurs when the selector is an identifier and not a class.

**Comment by twister65** [comment]
* With options reset:true
![without sync before reset](https://user-images.githubusercontent.com/26852967/46258361-9d388e00-c4c9-11e8-913e-1e77a5bd878a.gif)

* With options reset:true, beforeReset:function(){sr.sync()} 
![with sync before reset](https://user-images.githubusercontent.com/26852967/46258366-c9540f00-c4c9-11e8-9993-d30a6e324018.gif)

**Comment by twister65** [comment]
Otherwise, replace : 
https://github.com/scrollreveal/scrollreveal/blob/2f70956cca0a65afdd808d3036f3f111fd564862/dist/scrollreveal.js#L1043
by : 
`styles.push(element.styles.opacity.computed);`
And replace : 
https://github.com/scrollreveal/scrollreveal/blob/2f70956cca0a65afdd808d3036f3f111fd564862/dist/scrollreveal.js#L858
by : 
`styles.push(element.styles.opacity.computed);`
in initialize and triggerReset functions

**Comment by jlmakes** [comment]
I'm noticing in both examples, there are rather large images being revealed.

It's starting to sound like ScrollReveal is being initialized before these large images have finished loading. If that were true, then the element dimensions could be incorrect, making elements reveal/reset at unexpected times— and explain why calling `sync()` would fix things.

@camsarmiento try putting wrapping your ScrollReveal code in a `load` event handler, e.g:

```js
window.addEventListener('load', function() {
  ScrollReveal().reveal('.items');
});
```

**Comment by twister65** [comment]
Wrapping code for an article body : 
```
window.addEventListener('load', function() {
     sr.reveal('.main_component', { origin:'top', distance: '50px', duration:1330, reset:false });
});
```
Result : the article has disappeared
![agir pour la planete - avenir durable - google chrome 02_10_2018 17_38_58](https://user-images.githubusercontent.com/26852967/46360743-5c2caf00-c66c-11e8-91fa-e44a1887577a.gif)

**Comment by jlmakes** [comment]
Any updates @camsarmiento @JohnPhilosopher ?

**Comment by JohnPhilosopher** [comment]
> Any updates @camsarmiento @JohnPhilosopher ?

I can't really seem to come to a definitive conclusion as seemingly resizing the browser window in chrome fixes the problem, while it works properly in other browsers. I admittedly haven't tested much else after, but I'll look into it more.

**Comment by saintplay** [comment]
I agree with @jlmakes, is the responsibility of the developer to be aware of these image loading inconsistencies. Other libraries like [scroll-to-element](https://github.com/willhoag/scroll-to-element) that rely on element position will also fail, because of the same problem
